src/Gui/InputHintPy.cpp

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/****************************************************************************
+ *                                                                          *
+ *   Copyright (c) 2025 Kacper Donat <kacper@kadet.net>                     *
+ *                                                                          *
+ *   This file is part of FreeCAD.                                          *
+ *                                                                          *
+ *   FreeCAD is free software: you can redistribute it and/or modify it     *
+ *   under the terms of the GNU Lesser General Public License as            *
+ *   published by the Free Software Foundation, either version 2.1 of the   *
+ *   License, or (at your option) any later version.                        *
+ *                                                                          *
+ *   FreeCAD is distributed in the hope that it will be useful, but         *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU       *
+ *   Lesser General Public License for more details.                        *
+ *                                                                          *
+ *   You should have received a copy of the GNU Lesser General Public       *
+ *   License along with FreeCAD. If not, see                                *
+ *   <https://www.gnu.org/licenses/>.                                       *
+ *                                                                          *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+
+#include "InputHint.h"
+
+#include "InputHintPy.h"
+
+void Gui::registerUserInputEnumInPython(PyObject* module)
+{
+    using enum Gui::InputHint::UserInput;
+
+    constexpr const char* name = "UserInput";
+
+    PyObject* py_enum_class = NULL;
+    PyObject* py_enum_module = PyImport_ImportModule("enum");
+
+    if (py_enum_module == NULL) {
```

Comment:
Why check it against `NULL` instead of `nullptr` or just implicitly like we do in most other code:
```suggestion
    if (py_enum_module) {
```